src/Plaid/Link/OnSuccessMetadata.cs

```diff
@@ -0,0 +1,156 @@
+	/// in the Plaid Link flow is triggered. 
+	/// </summary>
+	public class OnSuccessMetadata
+	{
+
+		/// <summary>
+		/// The public_token is passed to the onSuccess Javascript callback as a separate parameter from 
+		/// the metadata structure itself. It must be manually added to the metadata in order to be 
+		/// included when deserializing.<br/>
+		/// <example>
+		/// <code>
+		///  onSuccess: async function(public_token, metadata)
+		///  {
+		///	   // The metadata object contains info about the institution the
+		///	   // user selected and the account ID or IDs, if the
+		///	   // Select Account view is enabled.
+		///	   // https://plaid.com/docs/#onsuccess-callback
+		///
+		///	   metadata.public_token = public_token;
+		///	  
+		///	   ... send the metadata as JSON to be deserialized into this class ...  
+		///	  
+		///	  }
+		/// </code>
+		/// </example>
+		/// </summary>
+		/// <value>The public token.</value>
+		[JsonProperty("public_token")]
+		public string PublicToken { get; set; } = null!;
+
+		/// <summary>
+		/// Gets or sets the link session identifier.
+		/// </summary>
+		/// <value>The link session identifier.</value>
+		[JsonProperty("link_session_id")]
+		public string LinkSessionId { get; set; } = null!;
+
+		/// <summary>
+		/// Gets or sets the link session identifier.
+		/// </summary>
+		/// <value>The link session identifier.</value>
+		[JsonProperty("account_id")]
+		public string AccountId { get; set; } = null!;
+
+		/// <summary>
+		/// Gets or sets the accounts info.
+		/// </summary>
+		/// <value>The accounts.</value>
+		[JsonProperty("accounts")]
+		[System.Diagnostics.CodeAnalysis.SuppressMessage(
+			"Performance",
+			"CA1819:Properties should not return arrays",
+			Justification = "Matching the data structure from the Plaid API")]
+		public AccountInfo[] Accounts { get; set; } = null!;
+
+		/// <summary>
+		/// The account object is not null in the onSuccess callback when 
+		/// your users connect a specific bank account via the Select Account 
+		/// pane in Link. You can enable this behavior in the Plaid Dashboard 
+		/// or by integrating with all Auth features.
+		/// 
+		/// Note: If Select Account is enabled for multiple accounts, account 
+		/// will contain only the first account. All accounts selected by the 
+		/// user will be present in the accounts object.
+		/// </summary>
+		[JsonProperty("account")]
+		public AccountInfo Account { get; set; } = null!;
+
+		/// <summary>
+		/// Gets or sets the institution.
+		/// </summary>
+		/// <value>The institution.</value>
+		[JsonProperty("institution")]
+		public InstitutionInfo Institution { get; set; } = null!;
+
+
+		/// <summary>
+		/// Represents an <see cref="Entity.Account"/> metadata.
+		/// </summary>
+		[System.Diagnostics.CodeAnalysis.SuppressMessage(
+			"Design",
+			"CA1034:Nested types should not be visible",
+			Justification = "Required for Json deserialization?")]
+		public class AccountInfo
```

Comment:
This appears to be a duplicate of the existing `Account` class. Any reason for this specifically?

Reply:
It's a subset of the properties plus an Id property. I know... friggin' annoying they didn't just use the same data structure